plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/tools/PostgreBackupWizardPageObjects.java

```diff
@@ -308,24 +308,28 @@ public void saveState() {
 
         List<PostgreDatabaseBackupInfo> objects = wizard.getSettings().getExportObjects();
         objects.clear();
-        List<PostgreSchema> schemas = new ArrayList<>();
-        List<PostgreTableBase> tables = new ArrayList<>();
+        List<PostgreSchema> entireSchemas = new ArrayList<>();
         for (TableItem item : schemasTable.getItems()) {
             if (item.getChecked()) {
                 PostgreSchema schema = (PostgreSchema) item.getData();
                 Set<PostgreTableBase> checkedTables = checkedObjects.get(schema);
-                // All tables checked
-                if (!schemas.contains(schema)) {
-                    schemas.add(schema);
-                }
-                if (checkedTables != null) {
-                    // Only a few tables checked
-                    tables.addAll(checkedTables);
+                if (checkedTables == null) {
+                    // All tables checked, we can add this schema in the full dump
+                    entireSchemas.add(schema);
+                    continue;
```

Comment:
It's a personal taste thingy, but I'd prefer to use `else` instead to ease understanding code flow.